tests/test_registration_retractions.py

```diff
@@ -0,0 +1,217 @@
+        self.justification = 'I loathe openness...'
+
+    def test_retract_private_registration_raises_400(self):
+        self.registration.is_public = False
+        self.registration.save()
+
+        res = self.app.post_json(
+            self.retraction_url,
+             auth=self.auth,
+             expect_errors=True,
+        )
+
+        assert_equal(res.status_code, 400)
+        self.registration.reload()
+        assert_false(self.registration.is_retracted)
+        assert_is_none(self.registration.retracted_justification)
+
+    def test_non_admin_retract_raises_401(self):
+        res = self.app.post_json(self.retraction_url, expect_errors=True)
+
+        assert_equals(res.status_code, 401)
+        self.registration.reload()
+        assert_false(self.registration.is_retracted)
+        assert_is_none(self.registration.retracted_justification)
+
+    def test_retract_without_justification_raises_200(self):
+        res = self.app.post_json(
+            self.retraction_url,
+             {'justification': ''},
+             auth=self.auth,
+        )
+
+        assert_equal(res.status_code, 200)
+        self.registration.reload()
+        assert_true(self.registration.is_retracted)
+        assert_equal(self.registration.retracted_justification, u'')
+
+    def test_redirect_if_retracted(self):
+        expected_redirect_url = self.registration.web_url_for('view_project')
+        res = self.app.post_json(
+            self.retraction_url,
+             {'justification': self.justification},
+             auth=self.auth,
+             expect_errors=True,
+        )
+
+        assert_equal(res.status_code, 200)
+        assert_equal(res.json['redirectUrl'], expected_redirect_url)
+        self.registration.reload()
+        assert_true(self.registration.is_retracted)
+        assert_equal(self.registration.retracted_justification, self.justification)
+
+    def test_cant_access_non_approved_resource(self):
+        # Retract public registration
+        expected_redirect_url = self.registration.web_url_for('view_project')
+        res = self.app.post_json(
+            self.retraction_url,
+             {'justification': self.justification},
+             auth=self.auth,
+             expect_errors=True,
+        )
+        # Verify it's accessible
+        assert_equal(res.status_code, 200)
+        assert_equal(res.json['redirectUrl'], expected_redirect_url)
+        self.registration.reload()
+        assert_true(self.registration.is_retracted)
+        # Ensure access to resources not explicitly granting permission to retractions returns 400
+        res = self.app.get(
+            self.registration.web_url_for('project_wiki_home'),
+            expect_errors=True
+        )
+        assert_equal(res.status_code, 400)
+
+
+class RegistrationRetractionSearchTestCase(OsfTestCase):
+
+    def test_retract_wiki_not_in_search(self):
+        pass
```

Comment:
Don't allow unfinished tests to pass silently. Stub them with something like `assert False, 'finish me'`. Use `unittest.skip` if you can't get to writing the test in the near future; otherwise, just fill out the test.